test/images/audit-proxy/main.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	auditinstall "k8s.io/apiserver/pkg/apis/audit/install"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	"k8s.io/apiserver/pkg/audit"
+)
+
+var (
+	encoder runtime.Encoder
+	decoder runtime.Decoder
+)
+
+func main() {
+	scheme := runtime.NewScheme()
+	auditinstall.Install(scheme)
+	serializer := json.NewSerializer(json.DefaultMetaFactory, scheme, scheme, false)
+	encoder = audit.Codecs.EncoderForVersion(serializer, auditv1.SchemeGroupVersion)
+	decoder = audit.Codecs.UniversalDecoder(auditv1.SchemeGroupVersion)
+
+	http.HandleFunc("/", handler)
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+func handler(w http.ResponseWriter, req *http.Request) {
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		log.Fatalf("could not read request body: %v", err)
```

Comment:
Do you want to terminate the process if it gets an error reading the request body, or should it instead return a 500 internal server error to the client?

Reply:
It would probably be most obvious to log and return a 500, updating